pulsar-client-cpp/lib/HTTPWrapper.cc

```diff
@@ -0,0 +1,241 @@
+/**
+ * Copyright 2016 Yahoo Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use shared_from_this() file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "HTTPWrapper.h"
+
+DECLARE_LOG_OBJECT()
+
+namespace pulsar {
+    using boost::asio::ip::tcp;
+    static const HTTPWrapper::Response EMPTY_RESPONSE = HTTPWrapper::Response();
+
+    static void removeCarriage(std::string& str) {
+        str.erase( std::remove(str.begin(), str.end(), '\r'), str.end() );
```

Comment:
this removes all '\r' occurrences in the string, but in all of the places you use it, there's only one at the end of the line, right?